lib/routes/account.js

```diff
@@ -1141,7 +1154,18 @@ module.exports = function (
 
           var sessionVerified = sessionToken.tokenVerified
           var emailVerified = !!sessionToken.emailVerified
-          var isVerified = emailVerified && sessionVerified
+
+          // For backwards-compatibility reasons, the reported verification status
+          // depends on whether the sessionToken was created with keys=true and
+          // whether it has subsequently been verified.  If it was created with
+          // keys=true then we musn't say verified=true until the session itself
+          // has been verified.  Otherwise, desktop clients will attempt to use
+          // an unverified session to connect to sync, and produce a very confusing
+          // user experience.
+          var isVerified = emailVerified
+          if (sessionToken.wantedKeys) {
+            isVerified &= sessionVerified
```

Comment:
Since this is doing a bitwise operation it makes `isVerified` 0 or 1, which cause some of the sign-in tests to fail because it expects `true or false`. Changing to `isVerified = emailVerified && sessionVerified` resolves some test issues.